root/static/scripts/common/utility/isDateEmpty.js

```diff
@@ -7,12 +7,20 @@
  * later version: http://www.gnu.org/licenses/gpl-2.0.txt
  */
 
+declare type PartialDateObservableT = {
+  +day: KnockoutObservable<string | null>,
+  +month: KnockoutObservable<string | null>,
+  +year: KnockoutObservable<string | null>,
+};
+
+declare type PartialDateStringT = {
```

Comment:
I feel these type names are slightly more clear plural since the types themselves aren't strings/observables, the contained values are.
```suggestion
declare type PartialDateObservablesT = {
  +day: KnockoutObservable<string | null>,
  +month: KnockoutObservable<string | null>,
  +year: KnockoutObservable<string | null>,
};

declare type PartialDateStringsT = {
```